config_utilities/include/config_utilities/virtual_config.h

```diff
@@ -35,12 +35,6 @@
 
 #pragma once
 
-#include <functional>
```

Comment:
I think (at least most of these, I saw memory, utility and string below at a glance) should be included here following the `include what you use` paradigm. It's bad practice to rely on transitive headers and due to the include guards I don't think this will slow down compilation. Cpplint will complain otherwise (I have that setup by default).